docs/whats-new/release-candidate-1.md

```diff
@@ -0,0 +1,358 @@
+
+The app host now supports waiting for dependencies to be ready before starting another resource. This is useful when a resource sources take a long time to start up. The app host will wait for the resource to be ready before starting the dependent resource.
+
+```C#
+var builder = DistributedApplication.CreateBuilder(args);
+
+var queue = builder.AddRabbitMQ("rabbit");
+
+builder.AddProject<Projects.WebApplication1>("api")
+    .WithReference(queue).WaitFor(queue);
+
+builder.Build().Run();
+```
+
+When the app host starts, it will wait for the `rabbit` resource to be ready before starting the `api` resource:
+
+<video controls src="videos/waitfor.mp4" title="Title"></video>
+
+There are 2 methods exposed to wait for a resource:
+
+1. `WaitFor` - Wait for a resource to be ready before starting another resource.
+2. `WaitForCompletion` - Wait for a resource to complete before starting another resource.
+
+#### Resource Health Checks
+
+`WaitFor` uses health checks to determine if a resource is ready. If a resource does not have any health checks, the app host will wait for the resource to be in the "Running" state before starting the dependent resource.
+
+You can now attach health checks to resources. These health checks use the `IHealthCheck` interface from the `Microsoft.Extensions.Diagnostics.HealthChecks` namespace.
+
+```C#
+var builder = DistributedApplication.CreateBuilder(args);
+
+var catalogApi = builder.AddContainer("catalog-api", "catalog-api")
+                 .WithHttpEndpoint(targetPort: 8080)
+                 .WithHttpHealthCheck("/healthz");
+
+builder.AddProject<Projects.WebApplication1>("store")
+    .WithReference(catalogApi.GetEndpoint("http"))
+    .WaitFor(catalogApi);
+
+builder.Build().Run();
+```
+
+The above example adds a health check to the `catalog-api` resource. The app host will wait for the health check to return a healthy status before starting the `store` resource. It'll determine that the resource is ready when the the /healthz endpoint returns a 200 status code.
+
+### Persistent Containers
+
+The app host now supports persistent containers. This is useful when you want to keep the container running even after the app host has stopped. These containers will not be stopped unless they are stopped manually using the container runtime.
+
+To do this, call the `WithLifetime` method and pass in `ContainerLifetime.Persistent`:
+
+```C#
+var builder = DistributedApplication.CreateBuilder(args);
+
+var queue = builder.AddRabbitMQ("rabbit")
+                   .WithLifetime(ContainerLifetime.Persistent);
+
+builder.AddProject<Projects.WebApplication1>("api")
+    .WithReference(queue).WaitFor(queue);
+
+builder.Build().Run();
+```
+
+The dashboard will show persistent containers with a pin icon:
+
+![Persistent containers](images/persistent-container.png)
+
+After the app host has stopped, the container will continue to run:
+
+![Docker desktop showing rabbit mq](images/persisent-container-dockerdesktop.png)
+
```

Comment:
Need some kind of qualification here about if/when we will restart the container.